src/Razor/test/Microsoft.VisualStudio.LanguageServices.Razor.Test/Cohost/TestTraceSourceProvider.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using Xunit.Abstractions;
+
+namespace Microsoft.VisualStudio.LanguageServices.Razor.Test.Cohost;
+
+/// <summary>
+/// An implementation of IServiceProvider that only provides a TraceSource, that writes to test output
+/// </summary>
+internal class TestTraceSourceProvider(ITestOutputHelper testOutputHelper) : IServiceProvider
```

Comment:
It's not worth changing anything, but I wanted to mention that there's a handy mock helper for creating `IServiceProvider`: `VsMocks.CreateServiceProvider(...)`.

Reply:
When its this easy to do, I'd rather avoid a mock 😛